src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Helpers.cs

```diff
@@ -12,25 +12,24 @@ public sealed partial class Utf8JsonWriter
     {
         private void ValidateWritingValue()
```

Comment:
Why not make this an aggressive-inlining method whose implementation is `if (!skip) { ValidateWritingValueSlowNonInlined(); }`? Then you don't have to change a dozen call sites.

Reply:
@GrabYourPitchforks Mono interpreter currently can't inline methods with branches, so it would no longer provide any benefit there.

Reply:
It does not feel right to be doing manual inlining and making the code less maintainable just to make it run faster under interpreter.

The interpreter is always go low throughput environment.

Reply:
Avoiding the overhead of calling a no-op method is a normal idiom.

Also the existing writer code already had the same up-front `if` check before calling `ValidateStart()` and `ValidateEnd()` (instead of within the called method) so one way to think about this is extending that existing pattern to `ValidateWritingValue`. Granted there were only 4 cases of that before and now there's 21.